lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -148,20 +148,24 @@ namespace Js
     uint32 ArrayBuffer::GetByteLengthFromVar(ScriptContext* scriptContext, Var length)
     {
         Var firstArgument = length;
+        double returnValue;
+
         if (TaggedInt::Is(firstArgument))
         {
-            int32 byteCount = TaggedInt::ToInt32(firstArgument);
-            if (byteCount < 0)
-            {
-                JavascriptError::ThrowRangeError(
-                    scriptContext, JSERR_ArrayLengthConstructIncorrect);
-            }
-            return byteCount;
+            returnValue = TaggedInt::ToInt32(firstArgument);
         }
         else
         {
-            return JavascriptConversion::ToUInt32(firstArgument, scriptContext);
+            returnValue = JavascriptConversion::ToNumber(firstArgument, scriptContext);
         }
+
+        if (returnValue < 0 || std::isinf(returnValue))
```

Comment:
@obastemur `IsFinite` is used all over the code base, it would be bad if it had a bug but I am skeptical since it has broad usage.  How did you test that it doesn't work and which build architecture did you test on?

Note I think it might also return false for NaN but I'm not certain.